process/block/metablock.go

```diff
@@ -1561,39 +1561,120 @@ func (mp *metaProcessor) ApplyBodyToHeader(hdr data.HeaderHandler, bodyHandler d
 	return nil
 }
 
-func (mp *metaProcessor) createEndOfEpochForMetablock(metaBlock *block.MetaBlock) (*block.EndOfEpoch, error) {
-	endOfEpoch := &block.EndOfEpoch{
-		PendingMiniBlockHeaders: make([]block.ShardMiniBlockHeader, 0),
-		LastFinalizedHeaders:    make([]block.FinalizedHeaders, 0),
+func (mp *metaProcessor) createEpochStartForMetablock(metaBlock *block.MetaBlock) (*block.EpochStart, error) {
```

Comment:
I think lines 1565-1567 + 1576-1604 can be extracted into a function getLastNotarizedAndFinalizedHeaders

Reply:
done.